docs/AddingNewProjects.md

```diff
@@ -0,0 +1,51 @@
+# Adding New Projects To The Repo
+
+Sample PR of final result: https://github.com/dotnet/aspnetcore/pull/41945
+
+## Creating a new project
+1. Create a new folder that will house your `.csproj` and other project-related files.
+2. (EXTREMELY IMPORTANT) Inside this new folder, make a new folder for the source files of your project. For regular functionality-adding project this will be called `src`. However, if you are adding a different kind of project, it will be called something more applicable (ex. `test/` for a test project).
+3. In VS, Add a new `Solution Folder` in the new folder from the first step.
+4. Create the project via the VS `Add` menu (select the folder -> right click -> Add -> follow the wizard).
+
+    **Note:** (Only applicable to `src/` projects) Depending on what kind of project you are creating, VS will create different files in your project. You might also want to add the following files:
+    - `PublicAPI.Shipped.txt`
+      - Lists publically visible APIs that are exported from your final compiled `.dll`.
+      - This only lists APIs that have already been shipped to customers and cannot be changed.
+      - There is an empty template at `eng/PublicAPI.empty.txt` for your reference. You can copy and rename the file to add it to your project. Make sure the name is exactly as shown above.
+    - `PublicAPI.UnShipped.txt`
+      - Lists publicly visible APIs that are exported from your final compiled `.dll`. If this is not configured properly, you will get build errors. VS will warn you though with green squiggly lines. If you see these squiggly lines, open the VS Quick Actions (CTRL + '.') and select the option to and it to the public API.
+      - This only lists APIs that have NOT already been shipped to customers. So, these can still change.
+      - There is an empty template at `eng/PublicAPI.empty.txt` for your reference. You can copy and rename the file to add it to your project. Make sure the name is exactly as shown above.
+    - `AssemblyInfo.cs`
+      - Lists various properties of your compiled assembly, such as to which other packages your `internal` properties and methods are available to.
+      - You can also expose internals via `@(InternalsVisibleTo)` items in your project file instead of using this `AssemblyInfo.cs` file.
+
+## Adding to the rest of the repo
+1. VS should have already registered your `.csproj` in the corresponding solution ([`.sln`](https://github.com/dotnet/aspnetcore/blob/586ccc8c895862b65645c4b0f979db1eecd29626/AspNetCore.sln)) and solution filter ([`.slnf`](https://github.com/dotnet/aspnetcore/blob/586ccc8c895862b65645c4b0f979db1eecd29626/src/Middleware/Middleware.slnf#L107-L109)) files.
```

Comment:
This is the first time the `.sln` or .`slnf` was mentioned. Why should they already be updated? Unless you opened an existing `.sln` or `.slnf` in the first place, I don't think it would be unless we're talking about a *new* `.sln` which we don't want.

The very start of this document should make it clear what we're opening in VS. If someone just does "Create a new project" outside the context of an existing solution, VS will helpfully (or unhelpfully in our case) try to crate a new `.sln` for you.

Reply:
I think the edit I suggested in https://github.com/dotnet/aspnetcore/pull/41949#discussion_r891641435 is probably sufficient for this.